man/s3fuse.1
.\" man page for s3fuse
.TH S3FUSE 1 2013-01-26 "s3fuse 0.13" "s3fuse"

.SH NAME
\fBs3fuse\fR - FUSE driver for cloud object storage services

.SH SYNOPSIS
.B  s3fuse
.RB [ \-f "] [" "\-h | \-\-help" "] [" \-o
.IR options ]
.RB [ "\-v | \-\-verbose" "] [" "\-V | \-\-version" ]
.I  mountpoint

.SH DESCRIPTION
\fBs3fuse\fR is a FUSE (Filesystem in Userspace) driver for Amazon S3 and 
Google Storage buckets with support for multipart transfers, file integrity
checks, transparent file encryption, and Glacier restores.

.SH OPTIONS
.TP
.B -f
Stay in the foreground -- don't daemonize/detach from the current session.

.TP
.B "-h | --help"
Display a summary of command line options.

.TP
.B "-v | --verbose"
Increase output verbosity. Can be repeated for more verbosity. Should be used 
with \fB-f\fR.

.TP
.B "-V | --version"
Print \fBs3fuse\fR version.

.TP
.BI -o " options"
Pass a comma-separated list of options to FUSE. \fIoptions\fR can be 
\fBs3fuse\fR-specific or generic FUSE options.  The \fBs3fuse\fR-specific 
options are:

.RS
.TP
.BI "config=" "file"
Read configuration from \fIfile\fR rather than the default locations
(~/.s3fuse/s3fuse.conf and /etc/s3fuse.conf).

.RE
.RS
Some generic FUSE options that may be useful are:

.TP
.BI "allow_other"
Allow users besides the one launching \fBs3fuse\fR to access the mounted file
system.

.TP
.BI "allow_root"
Allow root to access the mounted file system.

.TP
.BI "gid=" "group id"
Set apparent group ID for all files in the mounted bucket to \fIgroup id\fR.

.TP
.BI "uid=" "user id"
Set apparent user ID for all files in the mounted bucket to \fIuser id\fR.

.RE

.SH CONFIGURATION
See \fBs3fuse.conf\fR(5).

.SH AUTHORS
Tarick Bedeir <tarick@bedeir.com>

.SH SEE ALSO
\fBs3fuse.conf\fR(5), \fBs3fuse_gs_get_token\fR(1), \fBs3fuse_sha256_sum\fR(1),
\fBs3fuse_vol_key\fR(1)


ChangeLog
s3fuse (0.15) unstable; urgency=low

  * Removed known-issue notes on MIME mapping and Google Cloud Storage
    Manager folder representation incompatibility.

 -- Tarick Bedeir <tarick@bedeir.com>  Tue, 21 May 2013 21:25:12 -0600

s3fuse (0.14) unstable; urgency=low

  * NEW: Multipart (resumable) uploads for Google Cloud Storage.
  * NEW: Support for FV/S.
  * NEW: Set file content type by examining extension.
  * NEW: Set Cache-Control header with extended attribute.
  * NEW: Allow creation of "special" files (FIFOs, devices, etc.).
  * Allow verbosity to be specified explicitly on the command line.
  * Fix multipart download bug that prevented some transfers from resuming.
  * Use real user ID/group ID rather than effective IDs.
  * Only use user ID/group ID/mode in object metadata if values are valid.
  * Set and retrieve object creation time (on newly-created objects).
  * Fix inconsistent state bug when reopening file immediately after closing.
  * Various POSIX compliance fixes.

 -- Tarick Bedeir <tarick@bedeir.com>  Sat, 06 Apr 2013 14:57:10 -0600

s3fuse (0.13.1) unstable; urgency=low

  * OS X: Fixed bug that resulted in command-line arguments being ignored.

 -- Tarick Bedeir <tarick@bedeir.com>  Fri, 15 Feb 2013 20:06:26 -0600

s3fuse (0.13) unstable; urgency=low

  * NEW: File encryption. See README.
  * NEW: Glacier restore requests. See README.
  * NEW: Add least-recently-used eviction to object cache, controlled by the
    max_objects_in_cache configuration option.
  * NEW: OS X binaries.
  * IMPORTANT: Removed "auth_data" configuration option. For AWS, use
    "aws_secret_file" instead. For Google Storage, use "gs_token_file".
  * IMPORTANT: Search for configuration file in ~/.s3fuse/s3fuse.conf before
    trying %sysconfdir/s3fuse.conf (this is usually /etc/s3fuse.conf or
    /usr/local/s3fuse.conf).
  * Switched to SHA256 for file integrity check. File hash, if available, will 
    be in the "s3fuse_sha256" extended attribute.
  * Added standalone SHA256 hash generator ("s3fuse_sha256_hash") that uses the
    same hash-list method as s3fuse.
  * Changed handling of last-modified times. Some timestamps may be inaccurate 
    (but will be correct when updated).
  * Renamed built-in extended attributes "__content_type__" and "__etag__" to 
    "s3fuse_content_type" and "s3fuse_etag", respectively.
  * Removed "__md5__" extended attribute.
  * Added note about daemon_timeout.
  * Removed xattr_prefix configuration option and made "user." standard (Linux
    only).
  * Added event counters/statistics ("stats_file" config option).
  * Made "noappledouble" and "daemon_timeout=3600" default options on OS X.
  * Various robustness fixes.
  * Add manual pages.

 -- Tarick Bedeir <tarick@bedeir.com>  Sun, 10 Feb 2013 10:51:16 -0600

s3fuse (0.12) unstable; urgency=low

  * NEW: MacOS support. See README.
  * NEW: Made AWS endpoint configurable (to support non-US buckets). 
    See README.
  * Set mount point UID/GID to process effective UID/GID.
  * Fixed extended attribute bugs.
  * Added support for curl/NSS.
  * Fixed circular ownership bug that was causing FDs to be leaked.
  * Made requirements compatible with RHEL6/Amazon Linux.
  * Fixed memory leak in CURL request code.

 -- Tarick Bedeir <tarick@bedeir.com>  Sat, 08 Sep 2012 19:36:13 -0600

s3fuse (0.11) unstable; urgency=low

  * NEW: Added Google Storage support.
  * NEW: Moved AWS keys out of s3fuse.conf. See README.
  * NEW: Changed required fields in s3fuse.conf. See README.
  * On rename, remove destination rather than failing with -EEXIST.
  * Improved error handling in FUSE init/pre-init.
  * Invalidate parent in directory cache when child is modified.
  * Fixed request error handling.
  * Increased default request timeout to 30 seconds (from 10).
  * Changed default logging level to LOG_WARNING instead of LOG_ERR.

 -- Tarick Bedeir <tarick@bedeir.com>  Fri, 15 Jul 2011 19:45:59 -0500

s3fuse (0.10) unstable; urgency=low

  * Updated usage/help message printed by main() to match arguments.
  * Added directory cache. 

 -- Tarick Bedeir <tarick@bedeir.com>  Thu, 30 Jun 2011 18:14:25 -0500

s3fuse (0.9) unstable; urgency=low

  * Moved to libxml++.

 -- Tarick Bedeir <tarick@bedeir.com>  Sun, 12 Jun 2011 17:43:26 -0500

s3fuse (0.1) unstable; urgency=low

  * Initial Release.

 -- Tarick Bedeir <tarick@bedeir.com>  Wed, 01 Jun 2011 17:06:31 -0500


man/s3fuse.conf.5.sh
#!/bin/bash

_CONF=$1

if [ ! -f "$_CONF" ]; then
  echo "can't find config file [$_CONF]."
  exit 1
fi

echo '.\" man page for s3fuse'
echo '.\" generated by s3fuse.conf.5.in'

echo '.TH S3FUSE.CONF 5 2013-01-26 "s3fuse 0.13" "s3fuse Configuration File"'
echo ''
echo '.SH NAME'
echo '\fBs3fuse.conf\fR - Configuration file for s3fuse'
echo ''
echo '.SH FILES'
echo '\fBs3fuse\fR(1) loads a configuration file on initialization. It tries'
echo 'the following paths, in order:'
echo '.IP \[bu] 2'
echo '.B "~/.s3fuse/s3fuse.conf"'
echo ''
echo '.IP \[bu]'
echo ".B \"$SYSCONFDIR/s3fuse.conf\""
echo ''
echo '.SH SYNTAX'
echo 'The configuration file contains a list of parameters, one per line, in'
echo 'the format \fIparameter_name\fR=\fIparameter_value\fR. Lines beginning'
echo 'with a # are ignored.'

grep '^\(CONFIG\|CONFIG_REQUIRED\|CONFIG_SECTION\)(.*);$' $_CONF | \
  sed -e 's/(/%%%%/' -e 's/);$//' -e 's/, /%%%%/g' -e 's/"//g' | \
  awk -F '%%%%' '{ \
    if ($1 == "CONFIG_SECTION") { \
      print "\n.SH " toupper($2) \
    } else if ($1 == "CONFIG_REQUIRED" || $1 == "CONFIG") { \
      print ".TP\n.B " $3 "\n" $5 \
    } \
  }'

echo '.SH SEE ALSO'
echo '\fBs3fuse\fR(1), \fBs3fuse_gs_get_token\fR(1), \fBs3fuse_vol_key\fR(1)'
